### ISSUE TITLE
Test Content for Splicing Tests

### ISSUE THREAD

**Issue by louaybassbouss** [cta-wave/Test-Content#19]
Test Content for following tests is missing:

- 8.8 Playback over WAVE Baseline Splice Constraints
- 8.13 Restricted Splicing of Encrypted Content
- 8.14 Sequential Playback of Encrypted and Non-Encrypted Baseline Content
- 9.4 Splicing of WAVE Program with Baseline Constraints

**Comment by jpiesing** [comment]
> Test Content for following tests is missing:
> 
>     * 8.8 Playback over WAVE Baseline Splice Constraints
> 
>     * 8.13 Restricted Splicing of Encrypted Content
> 
>     * 8.14 Sequential Playback of Encrypted and Non-Encrypted Baseline Content
> 
>     * 9.4 Splicing of WAVE Program with Baseline Constraints

Which mezzanine content does this correspond to?

**Comment by rbouqueau** [comment]
Some first content:
- http://reference.dashif.org/dash.js/latest/samples/dash-if-reference-player/index.html?mpd=https://www.gpac-licensing.com/downloads/CTA-Wave/splice_clear_9.4/stream.mpd
- http://reference.dashif.org/dash.js/latest/samples/dash-if-reference-player/index.html?mpd=https://www.gpac-licensing.com/downloads/CTA-Wave/splice_clear_cenc_8.14/stream.mpd
- http://reference.dashif.org/dash.js/latest/samples/dash-if-reference-player/index.html?mpd=https://www.gpac-licensing.com/downloads/CTA-Wave/splice_cenc_8.13/stream.mpd

Any feedback welcome. Only the first stream (in clear) is playable. The manifests are very similar (some periods are in clear while others are encrypted). If it looks ok then I will update it to the Akamai server.

**Comment by rbouqueau** [comment]
Ping @louaybassbouss :)

**Comment by louaybassbouss** [comment]
@FritzHeiden FYI new splicing test content is now available.

**Comment by louaybassbouss** [comment]
@rbouqueau  the content is avi

> Some first splice content:
> 
> * http://reference.dashif.org/dash.js/latest/samples/dash-if-reference-player/index.html?mpd=https://www.gpac-licensing.com/downloads/CTA-Wave/splice_clear_9.4/stream.mpd
> * http://reference.dashif.org/dash.js/latest/samples/dash-if-reference-player/index.html?mpd=https://www.gpac-licensing.com/downloads/CTA-Wave/splice_clear_cenc_8.14/stream.mpd
> * http://reference.dashif.org/dash.js/latest/samples/dash-if-reference-player/index.html?mpd=https://www.gpac-licensing.com/downloads/CTA-Wave/splice_cenc_8.13/stream.mpd
> 
> Any feedback welcome. Only the first stream (in clear) is playable. The manifests are very similar (some periods are in clear while others are encrypted). If it looks ok then I will update it to the Akamai server.

@rbouqueau  can you provide each content in multiple MPDs (2 MPDs one for main and one for spliced) instead of multi-periods ? see issue https://github.com/cta-wave/dpctf-tests/issues/22 . 

We check the content manually the clear content is playing properly. Regarding cenc content, we were able to play it even if we don't pass the cenc  Key ID and key value. Can you check it please.

**Comment by rbouqueau** [comment]
@louaybassbouss Thanks for the feedback.

> Can you provide each content in multiple MPDs (2 MPDs one for main and one for spliced) instead of multi-periods ? see issue https://github.com/cta-wave/dpctf-tests/issues/22 .

Here are all the links. I'd be happy if we could clarify this (separate manifests vs multi-period manifest):
- Main content, clear: https://reference.dashif.org/dash.js/latest/samples/dash-if-reference-player/index.html?mpd=https://www.gpac-licensing.com/downloads/CTA-Wave/splice_main/2021-01-18/stream.mpd
- Main content, CENC: https://reference.dashif.org/dash.js/latest/samples/dash-if-reference-player/index.html?mpd=https://www.gpac-licensing.com/downloads/CTA-Wave/splice_main-cenc/2021-01-18/stream.mpd
- Ad content, clear: https://reference.dashif.org/dash.js/latest/samples/dash-if-reference-player/index.html?mpd=https://www.gpac-licensing.com/downloads/CTA-Wave/splice_ad/2021-01-18/stream.mpd
- Ad content, CENC: https://reference.dashif.org/dash.js/latest/samples/dash-if-reference-player/index.html?mpd=https://www.gpac-licensing.com/downloads/CTA-Wave/splice_ad-cenc/2021-01-18/stream.mpd

> Regarding cenc content, we were able to play it even if we don't pass the cenc Key ID and key value. Can you check it please.

See links above. I checked and the content is encrypted. The content still use the same keys, could that be why you can play it back?

**Comment by louaybassbouss** [comment]
> @louaybassbouss Thanks for the feedback.
> 
> > Can you provide each content in multiple MPDs (2 MPDs one for main and one for spliced) instead of multi-periods ? see issue [cta-wave/dpctf-tests#22](https://github.com/cta-wave/dpctf-tests/issues/22) .
> 
> Here are all the links. I'd be happy if we could clarify this (separate manifests vs multi-period manifest):
> 

The final decision was in [cta-wave/dpctf-tests#22](https://github.com/cta-wave/dpctf-tests/issues/22) to use multiple MPDs. There is no multi-period at all. 

> * Main content, clear: https://reference.dashif.org/dash.js/latest/samples/dash-if-reference-player/index.html?mpd=https://www.gpac-licensing.com/downloads/CTA-Wave/splice_main/2021-01-18/stream.mpd
> * Main content, CENC: https://reference.dashif.org/dash.js/latest/samples/dash-if-reference-player/index.html?mpd=https://www.gpac-licensing.com/downloads/CTA-Wave/splice_main-cenc/2021-01-18/stream.mpd
> * Ad content, clear: https://reference.dashif.org/dash.js/latest/samples/dash-if-reference-player/index.html?mpd=https://www.gpac-licensing.com/downloads/CTA-Wave/splice_ad/2021-01-18/stream.mpd
> * Ad content, CENC: https://reference.dashif.org/dash.js/latest/samples/dash-if-reference-player/index.html?mpd=https://www.gpac-licensing.com/downloads/CTA-Wave/splice_ad-cenc/2021-01-18/stream.mpd
> 

Thanks we will check and let you know

> > Regarding cenc content, we were able to play it even if we don't pass the cenc Key ID and key value. Can you check it please.
> 
> See links above. I checked and the content is encrypted. The content still use the same keys, could that be why you can play it back?

Thanks for the hint this can be the reason

**Comment by rbouqueau** [comment]
@louaybassbouss Note that the URLs (server) will evolve once this is validated on your side.

**Comment by louaybassbouss** [comment]
@rbouqueau the streams are working fine. Can you make them available via database.json? @jpiesing  regarding content durations 10s for main content and 5s for Ad content are fine from your side? From Test implementation perspective the duration doesn't matter.

**Comment by rbouqueau** [comment]
Agreement during the call to create a new "CFHD-splicing" database + front-end tab.

**Comment by louaybassbouss** [comment]
@rbouqueau when do you think can you make the new splicing content available in database.json?

**Comment by rbouqueau** [comment]
Added by 5ec1ca41dede7616dceb3c4d86e25cc7350d482c. Let me know if this is ok for you.

**Comment by louaybassbouss** [comment]
Thanks @rbouqueau . There is still a mismatch of the ID structure for few elements most are related ss1 streams (Example https://github.com/cta-wave/Test-Content/blob/master/database.json#L472) can you make all IDs Zip/MPD URLs following the same structure?

**Comment by rbouqueau** [comment]
Thanks for reporting. That's a bug in the script. The ss1 is handled separately. I should have a look this evening.

**Comment by rbouqueau** [comment]
That's fixed, thanks again. The script is safe, it was an error when merging the database.

**Comment by louaybassbouss** [comment]
@FritzHeiden  can we close this issue?

**Comment by FritzHeiden** [comment]
> @FritzHeiden can we close this issue?

Yes. For now we are using the provided splice_main and splice_ad content

**Comment by rbouqueau** [comment]
I arbitrarily copied the properties from T1.

**Comment by jpiesing** [comment]
> I arbitrarily copied the properties from T1.

Just now or when you made those streams?
Do splice_main and splice_ad_content need to be regenerated with the most recent changes?

**Comment by rbouqueau** [comment]
Today's notes: @rbouqueau evaluate the situation tomorrow after JCCP call + @nicholas-fr to check with fractional frame rate. If ok, then generate asap.

**Comment by rbouqueau** [comment]
See my comment in https://github.com/cta-wave/Test-Content-Generation/issues/55

**Comment by rbouqueau** [comment]
Content updated.

**Comment by gitwjr** [comment]
@FritzHeiden @louaybassbouss Fritz and Louay to check and then close if OK.

**Comment by gitwjr** [comment]
Closed. This was checked during the HbbTV plugfest.
